### DISCUSSION TITLE
Root path does not exist : Root plugin

### DISCUSSION THREAD

**Discussion by ssmntn210** [coredns/coredns#5317] [Q&A] [answered]
Hi everyone!
Was trying to use the file plugin, encountering the following error when checking the logs. Since using the file/root plugin is pretty straightforward, I wonder what is going wrong, if there is some other stuff I need to set up before using the plugin? The directory /etc/coredns/zones does exist for sure, and so does the file db.foo.example.com. I’m using a single node cluster.
```
[INFO] Reloading
[WARNING] plugin/root: Root path does not exist: /etc/coredns/zones
[WARNING] plugin/file: Failed to open "open /etc/coredns/zones/db.foo.example.com: no such file or directory": trying again in 1m0s
[INFO] plugin/reload: Running configuration SHA512 = <SHA512>
[INFO] Reloading complete
[ERROR] plugin/file: Failed to open zone "foo.example.com." in "/etc/coredns/zones/db.foo.example.com": open /etc/coredns/zones/db.foo.example.com: no such file or directory
```
Here is the ConfigMap for reference:
```
apiVersion: v1
data:
  Corefile: |
    foo.example.com.:53 {
        errors
        health
        root /etc/coredns/zones
        file db.foo.example.com
        prometheus :9153
        cache 5
        reload
        log
        loadbalance
    }
kind: ConfigMap
metadata:
  creationTimestamp: "2022-02-15T09:55:36Z"
  name: coredns 
  namespace: kube-system
  resourceVersion: "11034888"
  selfLink: /api/v1/namespaces/kube-system/configmaps/coredns
  uid: 6c6c69af-8e45-11ec-a27a-506b8d9bdf8f
```
Thanks in advance!

**Comment by chrisohaver** [comment, marked as answer]
In Kubernetes, have you created a ConfigMap for `db.foo.example.com`, and updated mounts in the coredns Deployment?
Without those, the file/path won't be created in the Pod.

**Reply by ssmntn210** [reply]
Thanks a lot! I had not done that, that seemed to be the problem. That being said, is possible to separate the key-value pair for db.foo.example.com from the ConfigMap for cordons? Asking because having the entire zone-file within the ConfigMap makes it a bit cumbersome to maintain, specially given the zone-file I would be working with will have a large number of entries.

**Reply by chrisohaver** [reply]
Yes, a separate ConfigMap can be attached as a separate `volume` with `volumeMounts` defined to mount it to a directory.

**Reply by ssmntn210** [reply]
Yes, I’ve tried it but getting the same error: . Here is what I did :
For volumeMounts:
```
volumeMounts:
        - name: config-volume-1
          mountPath: /etc/coredns/Corefile
          subPath: Corefile
        - name : config-volume-2
          mountPath: /etc/coredns/zones/db.foo.example.com
          subPath: db.example.com
```
For volumes:
```
volumes:
   - name: config-volume-1
    configMap:
     defaultMode: 420
     items:
     - key: Corefile
      path: Corefile
     name: coredns
   - name: config-volume-2
    configMap:
     defaultMode: 420
     items:
     - key: db.foo.example.com
      path: db.foo.example.com
     name: zone-file
```
(zone-file is the ConfigMap for the zone file)
I removed had made the root plugin to be "root zones" in the Corefile. Still getting the same, "open zones/db.foo.example.com: no such file or directory" error. Sorry, just started and getting used to Kubernetes...

**Reply by chrisohaver** [reply]
The mount path doesn’t match the path in your corefile.  /etc/coredns/zones != /zones.

**Reply by ssmntn210** [reply]
Thanks, that solves everything!
